chia/data_layer/data_store.py

```diff
@@ -2065,94 +1489,38 @@ async def remove_subscriptions(self, store_id: bytes32, urls: list[str]) -> None
-                pending_nodes AS (
-                    SELECT node_hash AS hash FROM root
-                    WHERE status IN (:pending_status, :pending_batch_status)
-                    UNION ALL
-                    SELECT n.left FROM node n
-                    INNER JOIN pending_nodes pn ON n.hash = pn.hash
-                    WHERE n.left IS NOT NULL
-                    UNION ALL
-                    SELECT n.right FROM node n
-                    INNER JOIN pending_nodes pn ON n.hash = pn.hash
-                    WHERE n.right IS NOT NULL
-                )
-
-                SELECT hash, left, right
-                FROM all_nodes
-                WHERE hash NOT IN (SELECT hash FROM ancestors WHERE tree_id != :tree_id)
-                AND hash NOT IN (SELECT hash from pending_nodes)
-                """,
-                {
-                    "tree_id": store_id,
-                    "pending_status": Status.PENDING.value,
-                    "pending_batch_status": Status.PENDING_BATCH.value,
-                },
-            )
-            to_delete: dict[bytes, tuple[bytes, bytes]] = {}
-            ref_counts: dict[bytes, int] = {}
-            async for row in cursor:
-                hash = row["hash"]
-                left = row["left"]
-                right = row["right"]
-                if hash in to_delete:
-                    prev_left, prev_right = to_delete[hash]
-                    assert prev_left == left
-                    assert prev_right == right
-                    continue
-                to_delete[hash] = (left, right)
-                if left is not None:
-                    ref_counts[left] = ref_counts.get(left, 0) + 1
-                if right is not None:
-                    ref_counts[right] = ref_counts.get(right, 0) + 1
-
-            await writer.execute("DELETE FROM ancestors WHERE tree_id == ?", (store_id,))
-            await writer.execute("DELETE FROM root WHERE tree_id == ?", (store_id,))
-            queue = [hash for hash in to_delete if ref_counts.get(hash, 0) == 0]
-            while queue:
-                hash = queue.pop(0)
-                if hash not in to_delete:
-                    continue
-                await writer.execute("DELETE FROM node WHERE hash == ?", (hash,))
-
-                left, right = to_delete[hash]
-                if left is not None:
-                    ref_counts[left] -= 1
-                    if ref_counts[left] == 0:
-                        queue.append(left)
-
-                if right is not None:
-                    ref_counts[right] -= 1
-                    if ref_counts[right] == 0:
-                        queue.append(right)
-
     async def unsubscribe(self, store_id: bytes32) -> None:
         async with self.db_wrapper.writer() as writer:
             await writer.execute(
                 "DELETE FROM subscriptions WHERE tree_id == :tree_id",
                 {"tree_id": store_id},
             )
+            await writer.execute(
```

Comment:
I thought we intentionally had separation of unsubscribing and deleting data.

Reply:
as far as I can remember, it wasn't intentional - it was a way to make delete data non blocking of other operations. but duplicating everything by store_id in this new schema solves this (we're deleting fast, as opposed to before where we'd waste time and block everything)

Reply:
maybe i'm thinking of the `.dat` deletion that's controllable via the cli `--retain` option.  alrighty.